### PR TITLE
Implement RFC 221 changes to ViewOnSourcegraphButton

### LINKED ISSUE #3669: "view on Sourcegraph" buttons should always open in new tabs
They open a different host, so I would always expect them to open in a new tab.
It is very annoying to lose your context, review state and codeintel connection state in the mid of a code review.

### LINKED ISSUE #14225: Improve the 'Configure Sourcegraph' flow on GitHub
Part of [RFC 221](https://docs.google.com/document/d/19f4xleYBU1zZZdqMmXlLmFxeR-fwEpOwTOgViOFOnyo/edit?usp=sharing): Improve discoverability/awareness of the browser extension

**Description**

- make the Sourcegraph logo in the button colorful (instead of grey) to make the button look actionable.
- open the browser extension settings after clicking on the button

**Design documentation**

1. Button visuals

Current:
Screenshot 2020-09-28 at 08 44 39

New:
Screenshot 2020-09-28 at 08 44 45

2. Open the browser extension settings
- currently, they are opened in the separate tab, open them from the top bar instead (just as when the user clicks on the extension icon). [Alert](https://www.figma.com/file/V5t7KOzC6PIX75L4dz8IFc/RFC-221-Browser-extension-improve-discoverability-and-awareness-Approved?node-id=397%3A524) about the private repo should be displayed in the setting window.

### PR DESCRIPTION
**Pull request by tjkandala**
Resolves https://github.com/sourcegraph/security-issues/issues/81
Resolves #3669
Resolves #14225
(1 change/issue tackled per commit)

#### Private github.com repositories issue
Update selectors for `repoHeaderHasPrivateMarker` check.

#### Always open Sourcegraph in new tab
`target` and `rel` are set to `_blank` and `noopener norefferer` by default respectively.

#### Improve 'Configure on Sourcegraph' flow
1 ) Only apply `filter: grayscale(100%)` to `SourcegraphIconButton` for "real" errors, not for private repository errors.
2) Chrome does not implement `openPopup` for [security reasons](https://bugs.chromium.org/p/chromium/issues/detail?id=436489#c103), so we can't implement this change. (edit: Implementing in separate PR for working browsers)

### CODE REVIEW

File: browser/src/shared/code-hosts/github/codeHost.ts [hunk 1]
```diff
@@ -290,8 +290,8 @@ export const githubCodeHost: CodeHost = {
     nativeTooltipResolvers: [nativeTooltipResolver],
     getContext: () => {
         const repoHeaderHasPrivateMarker =
-            !!document.querySelector('.repohead .private') ||
```
Comment (felixfbecker) [inline browser/src/shared/code-hosts/github/codeHost.ts:293]: **question**: Could this still be relevant for older GitHub enterprise instances? Last time we fixed it we intentionally left it in.

File: browser/src/shared/code-hosts/github/codeHost.ts [hunk 2]
```diff
@@ -290,8 +290,8 @@ export const githubCodeHost: CodeHost = {
...
-            !!document.querySelector('#js-repo-pjax-container h1 .octicon-lock')
+            !!document.querySelector('main h1 .octicon-lock') ||
+            !![...document.querySelectorAll('main h1 span')].find(span => span.textContent?.toLowerCase() === 'private')
```
Comment (felixfbecker) [inline browser/src/shared/code-hosts/github/codeHost.ts:294]: **issue**: These are very broad selectors, I'm worried that these could match other elements on the page.
When we last fixed this, I suggested some possible alternatives:

https://github.com/sourcegraph/sourcegraph/pull/12466#issuecomment-663585281

**Comment by felixfbecker** [comment]
> 2) Chrome does not implement openPopup for security reasons, so we can't implement this change.

We _could_ implement it for Firefox (and Chrome unstable), right?

File: browser/src/shared/code-hosts/github/codeHost.ts (hunk 1 above)
Comment (tjkandala) [reply browser/src/shared/code-hosts/github/codeHost.ts:293]: I brought it back _and_ strengthened the existing selectors. 

> A bit nervous about over-matching but I guess better over-match than under-match for this

I assume you still feel the same way about this as in #12466?

**Review by felixfbecker — APPROVED** [review]

**Pull request merged**